bin/SleepRayTune.py

```diff
@@ -57,20 +57,20 @@ def SleepRayTune(config, cfg: DictConfig, num_epochs, num_gpu):
 
 @hydra.main(config_path="../config", config_name="config")
 # model=m2, dataset=m2, training=m2 in terminal
-def main(cfg: DictConfig, num_epochs=100, gpus_per_trial=1):
+def main(cfg: DictConfig, num_epochs=200, gpus_per_trial=1):
 
     logger = logging.getLogger()
 
     # restrict the number for cuda
-    os.environ["CUDA_VISIBLE_DEVICES"] = "1,4,3"
+    os.environ["CUDA_VISIBLE_DEVICES"] = "1,4,3,2"
 
     # define the parameter space
     config = {
         "enc_conv_c": tune.grid_search([10, 15, 20, 25]),
-        "enc_kern": tune.grid_search([3, 5]),
-        "enc_hidden": tune.grid_search([64, 128]),
-        "lr": tune.loguniform([1e-5, 1e-2]),
-        "weight_decay": tune.loguniform([1e-6, 1e-3]),
+        "enc_kern": tune.grid_search([3]),
+        "enc_hidden": tune.grid_search([64, 128, 192, 256]),
+        "lr": tune.loguniform(1e-5, 1e-2),
+        "weight_decay": tune.loguniform(1e-6, 1e-3),
```

Comment:
@ismael-mendoza Do you think these ranges should be set in the config files? Or would that be overly complicated?

Reply:
I think this can be added to a config file, since our config files accept lists this should be pretty straightforward to incorporate. @zzhaozhe-profolio let me know if you have questions about how to do this

Reply:
Would it be better to do this with something like
```yaml
enc_hidden:
 - 64
 - 256
 - 2
```
to be called as `tune.grid_search(list(range(*enc_hidden)))`
instead of listing every value in config file.

Or maybe more generic as
```yaml
enc_hidden: "tune.grid_search(...)"
```
to be called as `eval(enc_hidden)`

Reply:
I suppose it depends how many settings of enc_hidden we want to test. I'd suggest starting with the simpler way (ie listing the values) and make it more automated if it seems like that will be helpful